### ISSUE TITLE
chromium version stuck on 114.0.5720.0?

### ISSUE THREAD

**Issue by J-Zeitler** [go-rod/rod#951]
Rod Version: v0.114.3

I noticed that the chromium registry used in Rod does not contain updates after 2023-04-21. This one being the latest (as far as I can see): https://registry.npmmirror.com/-/binary/chromium-browser-snapshots/Linux_x64/1133702/. ~That's also the default specified in https://github.com/go-rod/rod/blob/main/lib/launcher/revision.go#L6~ (on closer inspection: it's not exactly the same)

I'm investigating impact from https://nvd.nist.gov/vuln/detail/CVE-2023-4863. Would it be possible to use some other repository listing to utilize a more recent version of the browser?

For now I use the [custom launch](https://go-rod.github.io/#/custom-launch) option to mitigate the CVE in the meantime.

**Comment by ysmood** [comment]
Rod github action cron job to auto update revision.

I think some OS doesn't update the revision somehow, this is the algorithm:

https://github.com/go-rod/rod/blob/6b4ac725a434b4f08fad2f6446fbee877c0ef66a/lib/launcher/revision/main.go#L88

**Comment by J-Zeitler** [comment]
Saw that, I guess regardless how we query the mirror if they have not updated the registry we're just going to be stuck until they do:

https://github.com/go-rod/rod/blob/6b4ac725a434b4f08fad2f6446fbee877c0ef66a/lib/launcher/revision/main.go#L19-L22

On a similar note, the Playwright registry seems to be updated past the CVE above at least: https://raw.githubusercontent.com/microsoft/playwright/v1.38.1/packages/playwright-core/browsers.json

![image](https://github.com/go-rod/rod/assets/1719257/2b52118f-0a72-4979-a2be-c23414fceaf6)

**Comment by creativeprojects** [comment]
I believe it's also possible to get this information from https://chromiumdash.appspot.com/

Using this endpoint https://chromiumdash.appspot.com/fetch_releases?channel=Stable&platform=Linux&num=1&offset=0 returns:

```json
[
  {
    "channel": "Stable",
    "chromium_main_branch_position": 1181205,
    "hashes": {
      "angle": "be854a0aa2c5ca2a3c00d1ba1df4101dfb434556",
      "chromium": "e3344ddefa12e60436fa28c81cf207c1afb4d0a9",
      "dawn": "0bb4f2362aab025f0221188d70f8b352da8a83c2",
      "devtools": "785d618a9556d5b6e8b3423721a93391afb8e519",
      "pdfium": "df6fed9f3cbc13b656562b4775ea330c91732c71",
      "skia": "beb4d7d77fba1e7c44b82653090e176bac161d63",
      "v8": "1add2a3f621ccbce65daacc6067fe3b440884557",
      "webrtc": "70aa7e99e4af06e9a2273793179dfcfddad11898"
    },
    "milestone": 117,
    "platform": "Linux",
    "previous_version": "117.0.5938.132",
    "time": 1696445220000,
    "version": "117.0.5938.149"
  }
]
```

**Comment by ysmood** [comment]
@creativeprojects Nice catch!

**Comment by ysmood** [comment]
The problem is where to download them, the npm registry also provide binary download link.

**Comment by creativeprojects** [comment]
The number in the field `chromium_main_branch_position` is the one called `Revision` in `*launcher.Browser`.
I have a script that always download the latest version using this 👍🏻

**Comment by ysmood** [comment]
Could you show us the script?

**Comment by creativeprojects** [comment]
🤔 you're right, there's something wrong.

It totally worked last week with version 117, but it doesn't work today with version 118 😢

**Comment by creativeprojects** [comment]
Right, I was looking at this again this morning and I found out that google is now releasing testing versions at the same time as the full version. Apparently the testing version doesn't include auto-update, and maybe other things.

It's easier to find the binary because it's actually using the version tag:

https://googlechromelabs.github.io/chrome-for-testing/
